frontend/packages/operator-lifecycle-manager/src/components/descriptors/spec/index.tsx

```diff
@@ -90,42 +90,68 @@ const BasicSelector: React.SFC<SpecCapabilityProps> = ({ value, capability }) =>
 );
 
 const BooleanSwitch: React.FC<SpecCapabilityProps> = (props) => {
+  const { model, obj, descriptor, onHandleError } = props;
   const convertedValue = !_.isNil(props.value) ? props.value : false;
   const [value, setValue] = React.useState(convertedValue);
   const [confirmed, setConfirmed] = React.useState(false);
+  const [errorMessage, setErrorMessage] = React.useState(null);
+
+  const errorCb = (err: Error): void => {
+    setConfirmed(false);
+    setValue(convertedValue);
+    setErrorMessage(err.message);
+    onHandleError(err.message);
+  };
 
-  const patchFor = (val: boolean) => [
-    { op: 'add', path: `/spec/${props.descriptor.path.replace('.', '/')}`, value: val },
-  ];
   const update = () => {
     setConfirmed(true);
-    return k8sPatch(props.model, props.obj, patchFor(value));
+    setErrorMessage(null);
+
+    const patchFor = (val: boolean) => [
+      { op: 'add', path: `/spec/${descriptor.path.replace(/\./g, '/')}`, value: val },
+    ];
+
+    let newObj = _.cloneDeep(obj);
+    newObj = _.set(newObj, `spec.${descriptor.path}`, value);
```

Comment:
We should only clone the object if we're going to use it, so this should move below the `if`.

`_.set` mutates `newObj`, so no need to resassign.

```suggestion
    const newObj = _.cloneDeep(obj);
    _.set(newObj, `spec.${descriptor.path}`, value);
```

Reply:
@spadgett Got it! Thanks.